### PR TITLE
fix: don't throw error in exchangeCodeForSession

### LINKED ISSUE #782: `exchangeCodeForSession` throws error instead of returning
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com).

## Describe the bug

When users log in using PKCE and click the email login link in a different browser, you get this `AuthApiError`:
> invalid request: both auth code and code verifier should be non-empty`

I believe that's expected, but the error is thrown instead of returned from `exchangeCodeForSession()`.

## To Reproduce

1. Set up PKCE with an auth handler like this:
	```ts
	const code = event.url.searchParams.get('code')
	if (typeof code === 'string') {
		const { data, error } = await supabase.auth.exchangeCodeForSession(code).catch((e) => {
			console.log('Thrown error', e);
		});
	}
	```
2. 

## Expected behavior

Since it's a expected and fairly common error, this error should be returned so that it can be properly handled

## System information

- OS: macOS
- Browser: Brave
- Version of supabase-js: 2.33.1
- Version of Node.js: 18.17.1

### PR DESCRIPTION
**Pull request by kangmingtay**
## What kind of change does this PR introduce?
* Fixes #782 
* Supercedes #945 

## What is the current behavior?

Please link any relevant issues here.

## What is the new behavior?

Feel free to include screenshots if it includes visual changes.

## Additional context

Add any other context or screenshots.

### CODE REVIEW

File: src/GoTrueClient.ts [hunk 1]
```diff
@@ -580,33 +580,43 @@ export default class GoTrueClient {
...
+          body: {
+            auth_code: authCode,
+            code_verifier: codeVerifier,
+          },
+          xform: _sessionResponse,
+        }
+      )
+      await removeItemAsync(this.storage, `${this.storageKey}-code-verifier`)
+      if (error) {
+        throw error
```
Comment (J0) [inline src/GoTrueClient.ts:600]: Should we continue to return the error here?  Seems like we return the error in the [original version](https://github.com/supabase/auth-js/pull/946/files#diff-3522461172efd6058d6b8da62fc2d30d8b524d2b64894ea2c67218c52f7fdff5L598) so wanted to confirm it is intended

`{ data: { user: null, session: null, redirectType: null }, error }`

Let me know if there are considerations I've missed though

**Review by J0 — APPROVED** [review]
Thanks for picking this up! Should help with and potentially close a few issues

File: src/GoTrueClient.ts (hunk 1 above)
Comment (kangmingtay) [reply src/GoTrueClient.ts:600]: @J0 it will get caught in the catch block below and returned

Comment (J0) [reply src/GoTrueClient.ts:600]: Perfect, thanks for the clarification

**Pull request merged**